search/lib/api/provider.js

```diff
@@ -50,6 +50,85 @@ async function getProvider (request, response) {
   }
 }
 
+/**
+ * Get all the cloud holding collections for the provider in the request.
+ */
+async function getCloudProvider (request, response) {
+  try {
```

Comment:
Let's reduce the size of the catch block to only the parts that can throw. Add more as needed instead of having the entire function inside one large one.

Reply:
This is copied from the existing getProvider and modified. I agree with you on a more targeted try catch block in general, if you know where the throw could happen. In this case, I'm not sure where the throw could happen because one function calls another and then maybe calls something outside of our code base...  In order not to miss anything, I will have to try catch every function?